lib/common/tests/xml/pcmk__xe_set_id_test.c

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2024 the Pacemaker project contributors
+ *
+ * The version control history for this file may have further details.
+ *
+ * This source code is licensed under the GNU General Public License version 2
+ * or later (GPLv2+) WITHOUT ANY WARRANTY.
+ */
+
+#include <crm_internal.h>
+
+#include <crm/common/unittest_internal.h>
+#include <crm/common/xml_internal.h>
+
+static void
+null_node(void **state)
+{
+    pcmk__xe_set_id(NULL, "test_id");
+}
+
+static void
+null_format(void **state)
+{
+    xmlNode *test_xml = pcmk__xe_create(NULL, "test");
+
+    pcmk__assert_asserts(pcmk__xe_set_id(NULL, NULL));
+    pcmk__assert_asserts(pcmk__xe_set_id(test_xml, NULL));
+
+    /* We can't test an empty format string, because the G_GNUC_PRINTF flag
+     * causes gcc to throw a "zero-length gnu_printf format string" error. We
+     * never reach the assertion in pcmk__xe_set_id().
+     */
+
+    pcmk__xml_free(test_xml);
+}
+
+static void
+valid_id(void **state)
+{
+    // IDs that are already valid XML names
+    xmlNode *test_xml = pcmk__xe_create(NULL, "test");
+    const char *id = NULL;
+
+    // No ID set initially
+    pcmk__xe_set_id(test_xml, "test_id");
+    assert_string_equal(crm_element_value(test_xml, PCMK_XA_ID), "test_id");
```

Comment:
not necessary, but we do have `pcmk__xe_id()`

Reply:
ah yes